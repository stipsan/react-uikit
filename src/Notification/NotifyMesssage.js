import { Component, PropTypes } from 'react'
import cx from 'classnames'
import Icon from '../Icon'

export default class NotifyMesssage extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    icon: PropTypes.string,
    isSticky: PropTypes.bool,
    timeout: PropTypes.number,
    type: PropTypes.oneOf(['info', 'success', 'danger', 'warning', 'primary', false]),
    onClick: PropTypes.func, // eslint-disable-line react/require-default-props
  }

  static defaultProps = {
    icon: '',
    isSticky: false,
    timeout: 3000,
    type: false,
  }

  state = {
    isOpen: false,
  }

  componentDidMount() {
    setTimeout(this.openNotification, 100)

    if (this.props.timeout && !this.props.isSticky) {
      setTimeout(this.handleClose, this.props.timeout)
    }
  }

  openNotification = () => this.setState({ isOpen: true })

  handleClose = () => {
    this.setState({ isOpen: false }, () => setTimeout(() => this.setState({ isClosed: true }), 300))
  }

  render() {
    const { children, type, icon, onClick } = this.props
    const className = cx('uk-notification-message', {
      'uk-notification-message-info': type === 'info',
      'uk-notification-message-success': type === 'success',
      'uk-notification-message-danger': type === 'danger',
      'uk-notification-message-warning': type === 'warning',
      'uk-notification-message-primary': type === 'primary',
    })
    const styles = [
      { marginTop: '-64px' },
      { marginTop: '0px' },
    ]

    if (this.state.isClosed) {
      return false
    }

    const body = icon ? (
      <div onClick={onClick}>
        <Icon icon={icon} />
        &nbsp;
        {children}
      </div>
    ) : (
      <div onClick={onClick}>
        {children}
      </div>
  )

    return (
      <div
        className={className}
        style={{ overflow: 'hidden', transition: 'margin ease-out 300ms', ...styles[this.state.isOpen ? 1 : 0] }}
      >
        <a className="uk-notification-close uk-close uk-icon" onClick={this.handleClose} >
          <Icon icon="close" />
        </a>
        {body}
      </div>
    )
  }
}
