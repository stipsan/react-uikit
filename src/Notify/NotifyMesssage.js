import { Component, PropTypes } from 'react'
import cx from 'classnames'

export default class NotifyMesssage extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    icon: PropTypes.string,
    isSticky: PropTypes.bool,
    timeout: PropTypes.number,
    type: PropTypes.oneOf(['info', 'success', 'danger', 'warning', false]),
    onClick: PropTypes.func, // eslint-disable-line react/require-default-props
  }

  static defaultProps = {
    icon: '',
    isSticky: false,
    timeout: 3000,
    type: false,
  }

  state = {
    isOpen: false,
  }

  componentDidMount() {
    setTimeout(this.openNotification, 100)

    if (this.props.timeout && !this.props.isSticky) {
      setTimeout(this.handleClose, this.props.timeout)
    }
  }

  openNotification = () => this.setState({ isOpen: true })

  handleClose = () => {
    this.setState({ isOpen: false }, () => setTimeout(() => this.setState({ isClosed: true }), 300))
  }

  render() {
    const { children, type, icon, onClick } = this.props
    const className = cx('uk-notify-message', {
      'uk-notify-message-info': type === 'info',
      'uk-notify-message-success': type === 'success',
      'uk-notify-message-danger': type === 'danger',
      'uk-notify-message-warning': type === 'warning',
    })
    const styles = [
      { marginTop: '-64px' },
      { marginTop: '0px' },
    ]

    if (this.state.isClosed) {
      return false
    }
    return (
      <div
        className={className}
        style={{ overflow: 'hidden', transition: 'margin ease-out 300ms', ...styles[this.state.isOpen ? 1 : 0] }}
      >
        <a className="uk-close" onClick={this.handleClose} />
        <div onClick={onClick}>
          {icon && <span><i className={`uk-icon-${icon}`} />&nbsp;</span> }
          {children}
        </div>
      </div>
    )
  }
}
