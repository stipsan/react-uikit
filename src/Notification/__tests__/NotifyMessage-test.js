import renderer from 'react-test-renderer'

import NotifyMesssage from '../NotifyMesssage'

jest.useFakeTimers()

describe('NotifyMesssage', () => {
  const defaultProps = {
    children: 'children',
    icon: 'check',
    position: 'top-center',
    onClick: jest.fn(),
  }
  const component = renderer.create(
    <NotifyMesssage {...defaultProps} />
  )
  it('should render correctly', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should return if state is closed', () => {
    const instance = component.getInstance()
    instance.handleClose()
    jest.runAllTimers()
    expect(instance.state.isOpen).toBe(false)

    instance.openNotification()
    expect(instance.state.isOpen).toBe(true)
  })
})
