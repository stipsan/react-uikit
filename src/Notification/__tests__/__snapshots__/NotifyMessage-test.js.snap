// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotifyMesssage should render correctly 1`] = `
<div
  className="uk-notification-message"
  style={
    Object {
      "marginTop": "-64px",
      "overflow": "hidden",
      "transition": "margin ease-out 300ms",
    }
  }
>
  <a
    className="uk-notification-close uk-close uk-icon"
    onClick={[Function]}
  >
    <span
      className="uk-icon"
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"1.06\\" d=\\"M16,16 L4,4\\"></path><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"1.06\\" d=\\"M16,4 L4,16\\"></path>",
          }
        }
        height={20}
        viewBox="0 0 20 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      />
    </span>
  </a>
  <div
    onClick={[Function]}
  >
    <span
      className="uk-icon"
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<polyline fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"1.1\\" points=\\"4,10 8,15 17,4\\"></polyline>",
          }
        }
        height={20}
        viewBox="0 0 20 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      />
    </span>
    Â 
    children
  </div>
</div>
`;
